pennylane/measurements/mid_measure.py

```diff
@@ -82,6 +82,23 @@ def func():
         Python ``not``, ``and``, ``or``, do not work since these do not have dunder methods.
         Instead use ``~``, ``&``, ``|``.
 
+    Mid-circuit measurement results can also be returned from QNodes with :func:`sample` (finite 
+    shots) and :func:`expval` (finite shots and analytic).
```

Comment:
`counts`, `probs`, `var` should also work right?

Reply:
Ah yes good point!

Reply:
```suggestion
    Mid-circuit measurement results can also be returned from QNodes with finite shots or `shots=None` (analytic).
```

Reply:
How about something like
```
Mid-circuit measurement results can be processed with the usual measurement functions such as `qml.expval`. For QNodes with finite shots, `qml.sample` will return the actual sequence of mid-circuit measurements for all device executions.
```

Reply:
```suggestion
    Mid-circuit measurement results can be processed with the usual measurement functions such as
    :func:`~.expval`. For QNodes with finite shots, :func:`~.sample` applied to a mid-circuit measurement
    result will return a binary sequence of samples.
```
Slightly tweaking @vincentmr's suggestion.